plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java

```diff
@@ -451,4 +497,34 @@ public String getEncoding() {
             return encoding;
         }
     }
+    
+    class PanelEditorPaintListener implements PaintListener {
+
+        private String statusMessage;
+
+        public PanelEditorPaintListener() {
+            this.statusMessage = "";
+        }
+
+        public void updateStatusMessage(String statusMessage) {
+            this.statusMessage = statusMessage;
+        }
+
+        public void paintControl(PaintEvent event) {
+            Object source = event.getSource();
+            if (source instanceof Control) {
+                GC gc = event.gc;
+                Rectangle bounds = ((Control) source).getBounds();
+                gc.setForeground(UIUtils.getSharedColor(SharedTextColors.COLOR_WARNING));
+                final int height = gc.textExtent(statusMessage).y;
+                for (String line : statusMessage.split("\n")) {
+                    line = line.trim();
+                    Point ext = gc.textExtent(line);
+                    int x = bounds.x;
+                    int y = bounds.y + bounds.height - ext.y - height;
+                    gc.drawText(line, x, y);
+                }
+            }
+        }
+    }
```

Comment:
Why do we need to paint text over a control? Can't we just create a new label control?